### ISSUE TITLE
[Feature Request] Possibility of adding custom color schemes

### ISSUE THREAD

**Issue by developithecus** [chakra-ui/chakra-ui#6776]
### Description

Hello, I'd like to suggest adding the possibility of defining custom color schemes for components.

### Problem Statement/Justification

Basically, this would allow us to add styles based on the `colorScheme` provided. These styles should probably have higher priority than variant styles, as we probably want to override the default colors inside a variant by providing a different color scheme. This solution would prevent having to create a function for every variant just to handle all the different color schemes possible for a component (we could probably use it to override default color schemes as well?).

### Proposed Solution or API

I think the API could look something like this:

```js
const Button = {
    baseStyle: {},
    variants: {},
    // new colorSchemes key on the ComponentStyleConfig interface
    colorSchemes: {
        "my-custom-scheme": {
            color: 'my-custom-color.500',
            background: 'red.500',    // could even use different colors, why not?
        }
    },
    defaultProps: {
        colorScheme: 'my-custom-scheme',   // could even set my custom colorScheme as default for buttons
    },
};

// usage:
<Button colorScheme="my-custom-scheme">
    Label
</Button>
```

Basically, adding a new key named `colorSchemes` to the `ComponentStyleConfig` interface. This new object would contain key-value pairs, where:
* keys are the names of the custom color scheme
* values are objects containing the styles for that given color scheme

### Alternatives

I am aware we can have a function as a variant and we could have styles created based on the `colorScheme` prop; however, that function would not be small & pretty if you just want to change a couple of properties for only a couple of color schemes. 

In my opinion, this suggestion would allow for better maintainability.

**Comment by anubra266** [comment]
@developithecus you can achieve this with [layerStyles](https://chakra-ui.com/docs/styled-system/text-and-layer-styles#layer-styles)
